File: training/java/src/main/java/de/sots/cellarsandcentaurs/domain/model/Dice.java
package de.sots.cellarsandcentaurs.domain.model;

import java.util.Random;

public final class Dice {
    private static final Random RANDOM = new Random();
    private static final int D20_SIDES = 20;

    private Dice() {
    }

    public static DiceRoll rollD20() {
        return new DiceRoll(RANDOM.nextInt(D20_SIDES) + 1);
    }
}

record DiceRoll(int d20Roll) {
    boolean isCritical() {
        return d20Roll == 20;
    }
}

File: training/javascript/src/de/sots/cellarsandcentaurs/domain/model/Dice.js
export class DiceRoll {
    constructor(sides, value) {
        this.sides = sides;
        this.value = value;
    }
}

export class Dice {
    constructor(sides) {
        this.sides = sides;
    }

    roll() {
        const value = Math.floor(Math.random() * this.sides) + 1;
        return new DiceRoll(this.sides, value);
    }
}

export function rollD20() {
    const d20Roll = new Dice(20).roll();
    return d20Roll.value;
}
